src/nplinker/loader.py

```diff
@@ -427,35 +413,46 @@ def _load_genomics(self):
 
         The attribute of `self.bgcs` is set to the loaded BGC objects that have the Strain object
         added (i.e. `BGC.strain` updated). If a BGC object does not have the Strain object, it is
-        not added to `self.bgcs`.
+        not added to `self.bgcs`. For MIBiG BGC objects, only those in use are added to `self.bgcs`.
 
         The attribute of `self.gcfs` is set to the loaded GCF objects that have the Strain objects
         added (i.e. `GCF._strains` updated). This means only BGC objects with updated Strain objects
         (i.e. `self.bgcs`) can be added to GCF objects.
         """
         logger.debug("\nLoading genomics data starts...")
 
-        # Step 1: load all BGC objects
+        # Step 1: load antismash BGC objects & add strain info
         logger.debug("Parsing AntiSMASH directory...")
         antismash_bgcs = AntismashBGCLoader(self.antismash_dir).get_bgcs()
-        raw_bgcs = antismash_bgcs + self.mibig_bgcs
+        antismash_bgcs_with_strain, _ = add_strain_to_bgc(self.strains, antismash_bgcs)
+
+        # Step 2: load mibig BGC objects (having strain info)
+        if self._use_mibig:
+            self.mibig_bgcs = MibigLoader(self.mibig_json_dir).get_bgcs()
 
-        # Step 2: load all GCF objects
+        # Step 3: get all BGC objects with strain info
+        all_bgcs_with_strain = antismash_bgcs_with_strain + self.mibig_bgcs
+
+        # Step 4: load all GCF objects
         # TODO: create a config for "bigscape_cluster_file" and discard "bigscape_dir" and "bigscape_cutoff"?
         bigscape_cluster_file = (
             Path(self.bigscape_dir) / "mix" / f"mix_clustering_c0.{self._bigscape_cutoff:02d}.tsv"
         )
         raw_gcfs = BigscapeGCFLoader(bigscape_cluster_file).get_gcfs()
 
-        # Step 3: add Strain object to BGC
-        bgc_with_strain, _ = add_strain_to_bgc(self.strains, raw_bgcs)
+        # Step 5: add BGC objects to GCF
+        all_gcfs_with_bgc, _, _ = add_bgc_to_gcf(all_bgcs_with_strain, raw_gcfs)
 
-        # Step 4: add BGC objects to GCF
-        gcf_with_bgc, _, _ = add_bgc_to_gcf(bgc_with_strain, raw_gcfs)
+        # Step 6: get mibig bgcs and strains in use from GCFs
+        mibig_bgcs_in_use = []
```

Comment:
I'd put this below, in an `else` statement after `if self._use_mibig:`

Reply:
Good idea!